### PR TITLE
New quickstarts.

### PR DESCRIPTION
**Pull request by jmdobry**
@ace-n Sorry for the size of the PR. It should be a quick review. I added a few new quickstarts and brought the other quickstarts up to standard.

### CODE REVIEW

**Comment by ace-n** [comment]
Note: circleCI is failing, seemingly because `google-cloud/dns` doesn't exist.

@jmdobry is this something you can fix? (I'll review/comment on the PR in the meantime - but that way we can run the unit tests before merging.)

**Comment by jmdobry** [comment]
Fixed, I forgot to add `dns` to the `scripts/install` file.

File: pubsub/system-test/quickstart.test.js [hunk 1]
```diff
@@ -35,14 +37,18 @@ describe(`pubsub:quickstart`, () => {
     const expectedTopicName = `my-new-topic`;
 
     pubsubMock = {
-      createTopic: (_topicName) => {
+      createTopic: (_topicName, _callback) => {
         assert.equal(_topicName, expectedTopicName);
+        assert.equal(typeof _callback, 'function');
```
Comment (ace-n) [inline pubsub/system-test/quickstart.test.js]: :+1: We should add callback-checking to the standard, IMO.

@jmdobry Maybe create an issue and/or submit a mini-PR for it, assuming you agree? (Alternatively, I'm happy to do this myself if you'd prefer.)

File: resource/quickstart.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+'use strict';
+
+// [START resource_quickstart]
+// Imports the Google Cloud client library
+const Resource = require('@google-cloud/resource');
+
+// Your Google Cloud Platform project ID
+const projectId = 'YOUR_PROJECT_ID';
```
Comment (ace-n) [inline resource/quickstart.js]: :+1: I like this explicit style.

Maybe another thing to add to the standard? (IIRC we had a discussion on how to specify these across the team - but I don't recall what the outcome of that was - I assume it was this?)

File: resource/system-test/projects.test.js [hunk 3]
```diff
@@ -1,30 +1,30 @@
...
-        assert(Array.isArray(projects));
-        assert(projects.length > 0);
-        assert(console.log.calledWith('Found %d project(s)!', projects.length));
-        setTimeout(done, 2000);
-      });
-    });
+const cwd = path.join(__dirname, `..`);
+const cmd = `node projects.js`;
+
+describe(`resource:projects`, () => {
```
Comment (ace-n) [inline resource/system-test/projects.test.js]: I like this for now - but at some point we need to talk to the team about what language we should write the global system tests in.

(I'm in favor of Node or Python, but I may be biased. :stuck_out_tongue_winking_eye:)

File: resource/test/projects.test.js [hunk 4]
```diff
@@ -1,77 +1,37 @@
...
+const proxyquire = require(`proxyquire`).noCallThru();
+
+describe(`resource:projects`, () => {
+  it(`should handle errors`, () => {
+    const error = new Error(`error`);
+    const callback = sinon.spy();
+    const resourceMock = {
+      getProjects: sinon.stub().yields(error)
+    };
+    const ResourceMocket = sinon.stub().returns(resourceMock);
```
Comment (ace-n) [inline resource/test/projects.test.js]: Is _Mock**et**_ a typo?

File: translate/system-test/quickstart.test.js [hunk 5]
```diff
@@ -18,21 +20,27 @@ const translate = proxyquire(`@google-cloud/translate`, {})({
   key: process.env.TRANSLATE_API_KEY
 });
 const string = `Hello, world!`;
+const translated = `Привет мир!`;
```
Comment (ace-n) [inline translate/system-test/quickstart.test.js]: Nit: rename to `expectedTranslation` or something like that? (i.e. denote that this is the *expected* result somehow.)

File: pubsub/system-test/quickstart.test.js (hunk 1 above)
Comment (jmdobry) [inline pubsub/system-test/quickstart.test.js]: Go ahead and open an issue.

File: dns/test/quickstart.test.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    DNSMock = sinon.stub().returns(dnsMock);
+  });
+
+  it(`should handle error`, () => {
+    proxyquire(`../quickstart`, {
+      '@google-cloud/dns': DNSMock
+    });
+
+    assert.equal(DNSMock.calledOnce, true);
+    assert.deepEqual(DNSMock.firstCall.args, [{ projectId: 'YOUR_PROJECT_ID' }]);
```
Comment (jmdobry) [inline dns/test/quickstart.test.js]: I don't think that's necessary. Plus, the `YOUR_PROJECT_ID` stuff in these quickstarts is temporary. I'll be removing it once google/google-auth-library-nodejs#98 lands

File: dns/system-test/zones.test.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+describe(`dns:zones`, () => {
+  before((done) => {
+    dns.createZone(zoneName, {
+      dnsName: `${process.env.GCLOUD_PROJECT}.appspot.com.`
+    }, done);
+  });
+
+  after((done) => {
+    dns.zone(zoneName).delete(() => {
+      dns.zone('my-awesome-zone').delete(() => {
```
Comment (jmdobry) [inline dns/system-test/zones.test.js]: Oops, deleted

File: dns/system-test/quickstart.test.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+ * limitations under the License.
+ */
+
+'use strict';
+
+const proxyquire = require(`proxyquire`).noPreserveCache();
+const dns = proxyquire(`@google-cloud/dns`, {})();
+
+const uuid = require(`uuid`);
+const zoneName = `test-${uuid().substr(0, 13)}`;
```
Comment (jmdobry) [inline dns/system-test/quickstart.test.js]: Switched to `substring`

File: dns/test/zones.test.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+      getZones: sinon.stub().yields(error)
+    };
+    const DNSMock = sinon.stub().returns(dnsMock);
+    const program = proxyquire(`../zones`, {
+      '@google-cloud/dns': DNSMock
+    });
+
+    program.listZones(callback);
+
+    assert.equal(callback.callCount, 1);
```
Comment (jmdobry) [inline dns/test/zones.test.js]: tldr; sinon only has `calledOnce`, `calledTwice`, and `calledThrice`. If the spy is called more than that, you have to use `callCount`. To create this test I copied another test that called the spy 8 times.

File: resource/test/projects.test.js (hunk 4 above)
Comment (jmdobry) [inline resource/test/projects.test.js]: Yes

File: translate/system-test/quickstart.test.js (hunk 5 above)
Comment (jmdobry) [inline translate/system-test/quickstart.test.js]: Done

**Comment by ace-n** [comment]
LGTM.

**Pull request merged**